Scripts/script-TaniumMultiHash.yml

```diff
@@ -0,0 +1,184 @@
+commonfields:
+  id: d7e15b38-e402-4969-85b4-84958da63cc1
+  version: 59
+name: TaniumMultiHash
+script: |-
+  # Python template - reading arguments, calling a command, handling errors and returning results
+  import pandas as pd
+  from itertools import chain
+  def filter_values(contents):
+      df_contents = pd.DataFrame(contents)
+      file_name_cols = []
+      for col in list(df_contents.columns.values):
+          if 'File Name' in col:
+              file_name_cols.append(col)
+
+      dir_cols=[]
+      for col in list(df_contents.columns.values):
+          if 'Directory Path' in col:
+              dir_cols.append(col)
+
+      md5_cols=[]
+      for col in list(df_contents.columns.values):
+          if 'MD5' in col:
+              md5_cols.append(col)
+
+      sha256_cols=[]
+      for col in list(df_contents.columns.values):
+          if 'SHA256' in col:
+              sha256_cols.append(col)
+      needed_cols = ['Computer Name']
+      needed_cols.extend(dir_cols)
+      needed_cols.extend(file_name_cols)
+      needed_cols.extend(md5_cols)
+      needed_cols.extend(sha256_cols)
+
+
+      df_contents=df_contents[needed_cols]
+
+      df_dir=df_contents.melt(id_vars=['Computer Name'],value_vars=dir_cols,var_name='dir_seq',value_name='dir')[['Computer Name','dir']]
+      df_file=df_contents.melt(id_vars=['Computer Name'],value_vars=file_name_cols,var_name='file_seq',value_name='file')[['file']]
+      df_md5=df_contents.melt(id_vars=['Computer Name'],value_vars=md5_cols,var_name='md5_seq',value_name='md5')[['md5']]
+      df_sha256=df_contents.melt(id_vars=['Computer Name'],value_vars=sha256_cols,var_name='sha256_seq',value_name='sha256')[['sha256']]
+
+      RES_UNAV= "[current result unavailable]"
+      RES_UNAV2= "[results currently unavailable]"
+      NOT_FOUND = "No Matches Found"
+      df_merge = pd.concat([df_dir,df_file,df_md5,df_sha256],axis=1)
+      df_filtered = df_merge[(df_merge['file'] != RES_UNAV2 ) &
+                             (df_merge['file'] != RES_UNAV ) &
+                             (df_merge['file'] != NOT_FOUND) &
+                             (df_merge['file'] != "")]
+      return df_filtered.to_dict('records')
+
+  md5_array = demisto.args()["hash_array"]
+  complete_pct = demisto.args()["complete_pct"]
+  page_length = int(demisto.args()["page_length"])
+
+  if isinstance(md5_array,list):
+      pass
+  else:
+      md5_array=[md5_array]
+
+  paged_list = [md5_array[i:i+page_length] for i in range(0,len(md5_array),page_length)]
+  hash_type = demisto.args()["hash_type"]
+  if hash_type=="MD5":
+      param_hash ="fileMD5Hash"
+  elif hash_type=="SHA256":
+      param_hash = "fileSHA256Hash"
+  for page in paged_list:
+      cpu_col = "Computer Name;Computer ID"
+      file_cols = "Index Query File Details{{"+param_hash+"={0},limit=1}}"
```

Comment:
why do we have the limit?

Reply:
If the same hash is found multiple times on the same computer it will throw off the pandas dataframe table for output.  I really only care if the file is on the machine one time.